samples/sample_common/src/vaapi_utils_x11.cpp

```diff
@@ -51,114 +50,99 @@ X11LibVA::X11LibVA(void)
+            if (!m_va_dpy_render)
             {
-                m_x11lib.XCloseDisplay(m_display);
                 sts = MFX_ERR_NULL_PTR;
             }
         }
         if (MFX_ERR_NONE == sts)
         {
             if (!setenv("LIBVA_DRIVER_NAME", PROCESSING_DRIVER_NAME, 1)) {
-            va_res = m_libva.vaInitialize(m_va_dpy, &major_version, &minor_version);
-            sts = va_to_mfx_status(va_res);
-            if (MFX_ERR_NONE != sts) {
-                m_x11lib.XCloseDisplay(m_display);
-                }
+                va_res = m_libva.vaInitialize(m_va_dpy, &major_version, &minor_version);
+                sts = va_to_mfx_status(va_res);
                 unsetenv("LIBVA_DRIVER_NAME");
             } else {
                 sts = MFX_ERR_NOT_INITIALIZED;
             }
         }
         if (MFX_ERR_NONE == sts)
         {
-        if (!msdk_strcmp(PROCESSING_DRIVER_NAME, RENDERING_DRIVER_NAME)) {
-            // same driver
-            VAStatus        va_res        = VA_STATUS_SUCCESS;
-            VAConfigID      vpp_config_id = VA_INVALID_ID;
-            VAConfigAttrib  cfgAttrib;
-
-            m_va_dpy_render = m_va_dpy;
-
-            cfgAttrib.type = VAConfigAttribRTFormat;
-            m_libva.vaGetConfigAttributes(m_va_dpy_render,
-                                          VAProfileNone,
-                                          VAEntrypointVideoProc,
-                                          &cfgAttrib,
-                                          1);
-
-            va_res = m_libva.vaCreateConfig(m_va_dpy_render,
-                                            VAProfileNone,
-                                            VAEntrypointVideoProc,
-                                            &cfgAttrib,
-                                            1,
-                                            &vpp_config_id);
-
-            /* Create a context for VPP pipe */
-            va_res = m_libva.vaCreateContext(m_va_dpy_render,
-                                             vpp_config_id,
-                                             0,
-                                             0,
-                                             VA_PROGRESSIVE,
-                                             0,
-                                             0,
-                                             &m_contextID);
-        } else {
-            m_va_dpy_render = m_vax11lib.vaGetDisplay(m_display);
-            if (!m_va_dpy_render)
-            {
-                m_libva.vaTerminate(m_va_dpy);
-                m_x11lib.XCloseDisplay(m_display);
-                sts = MFX_ERR_NULL_PTR;
-            }
-            if (!setenv("LIBVA_DRIVER_NAME", RENDERING_DRIVER_NAME, 1)) {
-                va_res = m_libva.vaInitialize(m_va_dpy_render, &major_version, &minor_version);
-                sts = va_to_mfx_status(va_res);
-                if (MFX_ERR_NONE != sts)
+            if (!msdk_strcmp(PROCESSING_DRIVER_NAME, RENDERING_DRIVER_NAME)) {
```

Comment:
I believe we need to completely drop this processing/rendering drivers stuff. So, the behind the scenes story was that iHD did not support efficient on-screen rendering via X11, but i965 did. So, to optimized that we have used 2 drivers approach. However, now we are going to support iHD driver only, so I believe that all this dual driver approach could be dropped. @alariono : do you agree?

This should be a preceding patch I believe.

Reply:
@dvrogozh, Yes, agree lets drop dual driver approach

Reply:
Dropped, please check.